_posts/events-magic.md

```diff
@@ -0,0 +1,81 @@
+---
+layout: post
+title: "Someone says event, and magically, coupling goes away"
+author: Mauro Servienti
+synopsis: "I feel events are used too many times as hammer to dismantle coupling. Unfortunately, it's gold plating. It might look better, but it'll fire back in the long term and costs more."
+header_image: /img/posts/events-magic/header.jpg
+tags:
+- architecture
+---
+
+I know... I'm grumpy, and [I've been ranting lately](https://milestone.topics.it/2024/01/27/cqrs-and-es.html). I promise it's the last time—last famous words 😬
+
+The [Mongo DB Is Web Scale](https://youtu.be/b2F-DItXtZs?si=7oZfEm0udIHCrGZZ) cartoon always makes me roll on the floor laughing. If you have never seen it, those are 5 minutes well spent. Take a look:
+
+<iframe width="560" height="315" src="https://www.youtube.com/embed/b2F-DItXtZs?si=WES0g_W1kT0GFY7K" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen></iframe>
+
+It's twelve years old. MongoDB may not be the main character anymore, and maybe it wasn't even then. It's more about stubbornness and the conviction that tools blindly put to work can solve any problem. The same applies to many other concepts back in the day and nowadays, too. Tools are not, and can't be, [silver bullets](https://en.wikipedia.org/wiki/No_Silver_Bullet).
+
+Recently, I was chatting with my dear friend [Weronika Łabaj](https://twitter.com/weralabaj), and she said what then became the title of this article and also something that can easily replace the word MongoDB in the presented meme/cartoon:
+
+> Someone says event, and magically, coupling goes away.
+
+We discussed this ongoing trend: There is this belief that introducing events into systems' architecture will magically eliminate coupling. Or, adopting an event-driven architecture is sufficient so that coupling never appears.
+
+Paraphrasing the meme: events are web-scale.
+
```

Comment:
```suggestion

One could argue what it is that makes people freak out when thinking about coupling. That's an appropriate and to-the-point question. Coupling, per se, is neither good nor bad. It's a metric measuring how much two components are "connected." Before trying to eliminate as much coupling as possible blindly, we should understand the context because [not all changes are born equal, and the same is true for coupling; not all coupling is bad coupling](https://milestone.topics.it/2021/03/10/not-all-changes-are-born-equal.html).

```